sklearn/decomposition/kernel_pca.py

```diff
@@ -259,7 +262,7 @@ def inverse_transform(self, X):
         "Learning to Find Pre-Images", G BakIr et al, 2004.
         """
         if not self.fit_inverse_transform:
-            raise ValueError("Inverse transform was not fitted!")
+            raise NotFittedError("Inverse transform was not fitted.")
```

Comment:
Indeed. Those 2 distinct cases might justify the introduction of a specific sub-class, e.g.:

```python
class IncompleteFitError(NotFittedError):
    """"Exception raised by models with optional fit behaviors

    Some models implement optional methods that only work when
    the instance was fitted with specific constructor parameter values. 
    This exception is raised by those methods when the hyperparameter
    value were not set properly.

    """"
```

What do you think?